### ISSUE TITLE
Add a tutorial for getting the binding agents for advancing in the game

### ISSUE THREAD

**Issue by hhyyrylainen** [Revolutionary-Games/Thrive#5428]
It should probably trigger like around editor cycle 11-12 if the player doesn't have a nucleus already to hint the player that what they need to place to advance in the game.

**Comment by La-ryssa** [comment]
Doesn't early game tutorial already tell the player to aim for getting a nucleus?

**Comment by hhyyrylainen** [comment]
That was added extremely recently. Still the player might forget so it would be good to have reminder tutorial after generation 10 to remind the player to start seriously consider the nucleus in order to progress (if they haven't placed one already and kept it).

**Comment by La-ryssa** [comment]
Speaking of, I see older tutorials would get their path by exporting a `NodePath` variable, and assigning it to a `CustomWindow` variable, but new tutorials immediately export the `CustomWindow` itself. Is that preferable?

**Comment by hhyyrylainen** [comment]
Yes, code written before Godot 4 had to use the old approach. And we still have a ton of code using it. There's an open issue about converting as many of the old bits of code to the new approach as possible: https://github.com/Revolutionary-Games/Thrive/issues/4940 if anyone has any appetite to do a lot of mechanical repetitious changes.

**Comment by La-ryssa** [comment]
I take it the best way to check if a player has a nucleus is  `ref var cellProperties = ref entity.Get<CellProperties>();` and `if (cellProperties.IsBacteria)`?

**Comment by La-ryssa** [comment]
What is the best way to check if a cell has a nucleus/a nucleus was placed?

**Comment by La-ryssa** [comment]
I've taken this approach:

`            case TutorialEventType.MicrobeEditorOrganellePlaced:
            {
                var eventArgs = (OrganellePlacedEventArgs)args;
                var isNucleus = eventArgs.Definition.Name == "Nucleus";

                if (isNucleus)
                {
                    Inhibit();
                }

                break;
            }`

**Comment by La-ryssa** [comment]
We could make it so when the player places down a nucleus, then the tutorial will not show up, even if the player removes the nucleus. Or we could make it so if the player doesn't have a nucleus at all, even if they did before, they will get the tutorial, for which we could simply check if the player is not a bacteria, although I need some guidance on what the best practice is to access the player node/its properties.

I'm more inclined for the latter option, as a new player could place a nucleus just to test it, remove it and then forget about it.

**Comment by hhyyrylainen** [comment]
> What is the best way to check if a cell has a nucleus/a nucleus was placed?

There's already tutorial for upgradeable organelles which does something similar.

On top of that just need to check that the player doesn't undo the nucleus place (or later the player redoes it).

That way it should be good enough to track if the player has a nucleus.

**Comment by La-ryssa** [comment]
Editor undo event doesn't have args so afaik you can't check what organelle was undone. Plus, if the player deletes the organelle, there are no events for that. I feel to work this way these two would need to be implemented. I've made it so if the nucleus is ever placed at all, the tutorial will be inhibited. Opening a pull request, feel free to change the text for the tutorial. Cheers

**Comment by hhyyrylainen** [comment]
I don't see a problem with the undo events receiving an args so that it can be detected what organelle was undone.

I'd like to see that guarantee against undoing the nucleus. Also because if this goes through with minimal changes: https://github.com/Revolutionary-Games/Thrive/pull/5938 the tutorial would use the nucleus placing and immediately undoing it as a learning part, so that would make the nucleus tutorial totally break.

**Comment by La-ryssa** [comment]
> I don't see a problem with the undo events receiving an args so that it can be detected what organelle was undone.

Should I add that as part of my PR, or a separate one?

> the tutorial would use the nucleus placing and immediately undoing it as a learning part

Do you want me to implement this also? It'd be fairly easy

**Comment by hhyyrylainen** [comment]
> Should I add that as part of my PR, or a separate one?

You should add it to your PR to make it a complete feature.

> Do you want me to implement this also as part of the PR? It'd be fairly easy

No, because that is part of that other PR I mentioned so things would get pretty muddled up if you also made the change (in probably a slightly different way) in your PR.

**Comment by hhyyrylainen** [comment]
First part of this is done by this PR: https://github.com/Revolutionary-Games/Thrive/pull/5972

**Comment by La-ryssa** [comment]
Would you like this tutorial to just be a reminder on how to place the binding agent, much like the nucleus, or for it to teach how the binding system works when you place the organelle and when you exit the editor?

**Comment by hhyyrylainen** [comment]
Reminding about placing the binding agents. I think the tooltip and hotbar button that appear are clear enough that a player who places the binding agent should be able to figure out how to use it by that point in the game.

**Comment by ender-titan1** [comment]
So what exactly should be the behaviour when the player does not have either the nucleus or binding agents?

My current implementation just doesn't show the tutorial if the player does not have a nucleus by cycle X as the nucleus reminder tutorial already mentions the binding agents. But the tutorial could also be coded to trigger a few cycles _after_ the player evolves the nucleus but did not yet evolve binding agents. I'm personally leaning toward the latter.

**Comment by hhyyrylainen** [comment]
I think this tutorial should always trigger sometime after the player gets the nucleus as long as the player doesn't have the binding agents added. So yeah I agree with you that the latter idea sounds better.
